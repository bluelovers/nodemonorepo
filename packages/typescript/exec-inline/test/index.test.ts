import childProcess from 'child_process'
import process from 'process'
import spawnSync, { SpawnSyncRepresented } from '../index'

it('matches snapshot', () => {
  const info: { [_: string]: any } = {}

  jest
    .spyOn(childProcess, 'spawnSync')
    .mockImplementation((...args): SpawnSyncRepresented => {
      info.spawnSyncArgs = args
      return { status: 123 }
    })

  jest
    .spyOn(process, 'exit')
    .mockImplementation((...args) => {
      info.processExitArgs = args
    })

  const representative = spawnSync('arg0', 'arg1', 'arg2', 'arg3')
  info.representative = representative
  representative.exit()

  expect(info).toMatchSnapshot()
})
