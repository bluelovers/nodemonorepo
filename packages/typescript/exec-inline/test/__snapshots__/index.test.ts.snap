// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Object {
  "processExitArgs": Array [
    123,
  ],
  "representative": SpawnSyncRepresentative {
    "represented": Object {
      "status": 123,
    },
  },
  "spawnSyncArgs": Array [
    "arg0",
    Array [
      "arg1",
      "arg2",
      "arg3",
    ],
    Object {
      "stdio": "inherit",
    },
  ],
}
`;
